How-To/Create-a-New-Residential-Feature.md

```diff
@@ -0,0 +1,182 @@
+# How to Add a New Feature to the Residential Model
+
+The Cook County Assessor's Office manages features which are used predict assessed values for Cook County properties. Many of these are created by staff members, either by modifying data that the Assessor's office already possesses (e.g. location to different types of parcels) or data which has to be queried from other sources (i.e. distance to OpenStreetMap roads). This guide provides a template for each step of this process: downloading the raw data, transforming it into a usable structure, creating the view which will be referenced by the model, incorporating it in the model view, and running an updated model.
+
+It is important to note that for many features, not all of these steps will be needed. For example, some data will come in a clean format, and thus will not need a raw file stored in `etl/scripts-ccao-data-raw-us-east-1/`, or data may come from internal sources, requiring no extraction at all. All feature creation occurs in the [`data-architecture`](https://github.com/ccao-data/data-architecture) repository, and any changes to the model pipeline occurs in the [model-res-avm](https://github.com/ccao-data/model-res-avm/) repository.
+
+------------------------------------------------------------------------
+
+## Step 1: Extract and Load the Raw Data
+
+### Option A: Use a Python or R Script in the [etl Directory](https://github.com/ccao-data/data-architecture/tree/master/etl)
+
+Data extraction scripts are typically created in R or Python in the `etl/scripts-ccao-data-raw-us-east-1/` folder. In the circumstance that the file requires no cleaning afterwards, they can be placed directly in the `scripts-ccao-data-warehouse-us-east-1` folder. Data sources often contain information which is relevant for institutional knowledge, but are not useful for modeling. Keeping data in its raw form provides redundancy in case there are complications from the data transformation or the data source changes over time.
+
+-   Identify which folder the script should go in. Although folders are often self-explanatory, the location may switch throughout the feature creation process. For example, `spatial/spatial-environment-secondary_road.R` is created as a `spatial` feature, but during the transformation step (`proximity.dist_pin_to_secondary_roads.sql)`, it shifts to the `proximity` folder since the metric (distance) is in relation to PINs.
+
+-   Set the `AWS_S3_RAW_BUCKET` environment variable to an S3 bucket where output files will be stored:
+
+```         
+import os
+import boto3
+from dotenv import load_dotenv
+```
+
+```         
+library(arrow)
+library(aws.s3)
+```
```

Comment:
**[Suggestion, non-blocking]** We could add [language identifiers](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting) to these code blocks to instruct GitHub to add syntax highlighting.

    ```python
    import os
    import boto3
    from dotenv import load_dotenv
    ```

    ```r
    library(arrow)
    library(aws.s3)
    ```